### PR TITLE
initial Crowd Helm Chart based on Confluence Helm Chart

### PR DESCRIPTION
**Pull request by LouisZschaler**
This crowd chart was created based on the confluence chart.
Please let me know if any additional efforts are required.

### CODE REVIEW

**Comment by gjacobsatl** [comment]
Thanks for the contribution - I just noticed when diffing this against the current Confluence charts that the recent PRs haven't been included.
Mainly https://github.com/atlassian-labs/data-center-helm-charts/pull/98 which switched the nfs-permission-fixer and https://github.com/atlassian-labs/data-center-helm-charts/pull/102 though I'm guessing all of them since whenever this was originally created. Can we get those included?

**Comment by pathob** [comment]
Thanks @gjacobsatl and yeah, we were a bit slow. We will add the latest features of the other charts.

**Comment by LouisZschaler** [comment]
I adopted relevant changes from #98 and #102.

Any idea why the CLA is still pending?

**Comment by pathob** [comment]
Hi @nanux and @gjacobsatl 

There is definetly something wrong with your CLA bot. After new commits are pushed here, the CLA check is failing again. Only after I tried to resubmit it and got the following error message, the CLA check succeeded again.

image

But now it looks like we already need a rebase again...

**Comment by nanux** [comment]
> Hi @nanux and @gjacobsatl
> 
> There is definetly something wrong with your CLA bot. After new commits are pushed here, the CLA check is failing again. Only after I tried to resubmit it and got the following error message, the CLA check succeeded again.
> 
> image
> 
> But now it looks like we already need a rebase again...

Thank you for the report - did you fill the agreement on https://cla-external.services.atlassian.com/cla?

**Comment by pathob** [comment]
> > Hi @nanux and @gjacobsatl
> > There is definetly something wrong with your CLA bot. After new commits are pushed here, the CLA check is failing again. Only after I tried to resubmit it and got the following error message, the CLA check succeeded again.
> > image
> > But now it looks like we already need a rebase again...
> 
> Thank you for the report - did you fill the agreement on https://cla-external.services.atlassian.com/cla?

Yes :)

**Comment by LouisZschaler** [comment]
Rebase performed and CLA resigned.

**Comment by gorakong** [comment]
@pathob that's because the CLA was signed with an incorrect email address. It's looking for llzschaler@aservo.com (with 2 L's in the front), whilst it was signed with lzschaler@aservo.com.

**Comment by gorakong** [comment]
I can go ahead and fix the record for you. In the meantime, to unblock yourself, you can go ahead and merge the PR.

**Comment by pathob** [comment]
@gorakong

> @pathob that's because the CLA was signed with an incorrect email address. It's looking for [llzschaler@aservo.com](mailto:llzschaler@aservo.com) (with 2 L's in the front), whilst it was signed with [lzschaler@aservo.com](mailto:lzschaler@aservo.com).

Oh

> I can go ahead and fix the record for you.

Thank you!

> In the meantime, to unblock yourself, you can go ahead and merge the PR.

How do you mean that? We cannot do the merge as we are not part of Atlassian. And it looks like we somehow need a strategy to get this merged, We are now out-of-date with the base branch for the third time this week...

**Comment by gorakong** [comment]
@pathob np! sorry, what i meant was the maintainers should feel free to merge the PR when they're ready 😄

**Comment by pathob** [comment]
Hi @nanux and @gjacobsatl

can we please try again? Branch is up to date again, and CLA check successful. :smiley:

**Comment by pathob** [comment]
Hi @nanux and @gjacobsatl

sorry, I really don't want to annoy you, but the longer this PR is open, the more it will diverge again from the other charts. Is this PR ready to get merged?

**Review by nanux — CHANGES_REQUESTED** [review]
Thank you for the Crowd chart! Sorry, it took longer, we had to consider potential implications to the scope of the project. Crowd wasn't originally planned to be included in the first version of officially supported helm charts. But I don't want your contribution to be lost. For now, we are still planning to provide official helm chart support for Jira, Bitbucket and Confluence in the first milestone. But the nature of the helm chart repo allows us to collocated charts that are officially supported and in alpha testing if we provide enough notice to the users to distinguish the level of support.

I will follow up your PR with the disclaimer for the Crowd chart and also some other `values.yaml` files for quicker testing so we can set up CI on our side. Overall the PR looks good, just a few comments:

Can you please include the application version so we are consistent with other products?

Can you also share what scenarios and use cases you tested with the helm chart? I have tested some basic functionality and one issue I have noticed is that the JDBC connection parameters are not picked up in the setup - see my comment.

**Comment by nanux** [comment]
Sorry, could you please merge `master` to branch so we can run the Github Actions here (I've updated them so they can run on PR)?

@pathob @LouisZschaler could you please also have a look at the questions above? I wonder if you had success with database configuration.

I would like to merge the PR soon so it doesn't diverge from `master` too much.

**Comment by pathob** [comment]
Hi @nanux , thanks for your review and sorry for the small delay now from our site.

We are fine with the missing official support that you also mentioned in the linked ticket. For us it is currently more important to also have a Crowd chart that does not diverge too much from the others and that we may be able to make future contributions to.

We will add the app version and rebase to the latest master soon.

And yes, you got us. We also noticed now that the Crowd Docker image does not provide the environment variables to set the database so the database section does not work at all. And we are also not 100% sure about the other features of the Charts as we couldn't test all of them.

For now we will remove the database section again. Users can still use the web interface to setup the database - and we btw. are doing it by creating a Crowd config file using an init container. That's why we didn't notice.

File: src/main/charts/crowd/Chart.yaml
Comment (LouisZschaler) [reply src/main/charts/crowd/Chart.yaml]: done

File: src/main/charts/crowd/values.yaml
Comment (LouisZschaler) [reply src/main/charts/crowd/values.yaml]: That is correct. The Docker image doesn't provide the necessary endpoint. Should that ever change, the values and the db connectivity test could be part of this chart again.
We use additional init containers with an initial db dump to jump the setup.

**Comment by nanux** [comment]
@pathob thank you very much. Please let me know if you get around to implement the changes. If not, I can merge this and change it on our side.

**Comment by LouisZschaler** [comment]
Sorry, I missed to adjust the README.md, deleting the database values. Now it's actually clean.

I would recommend to merge it like this, and see if the Crowd Docker Image will have the database endpoint eventually. Our specialized solutions should not be part of a clean Crowd helm chart.

**Review by nanux — APPROVED** [review]

**Comment by nanux** [comment]
Thanks @pathob and @LouisZschaler for pushing through with this change.

**Pull request merged**
